test/tools/yulInterpreter/EVMInstructionInterpreter.cpp

```diff
@@ -476,7 +485,7 @@ bool EVMInstructionInterpreter::accessMemory(u256 const& _offset, u256 const& _s
 	{
 		u256 newSize = (_offset + _size + 0x1f) & ~u256(0x1f);
 		m_state.msize = max(m_state.msize, newSize);
-		return _size <= 0xffff;
+		return _size <= 0xffff && _offset <= u256(numeric_limits<size_t>::max());
```

Comment:
This is a bug fix change: Since we only interpret memory offsets modulo `numeric_limits<size_t>::max()`, not having this change would mean, we interpret

```
{
  let sizeTMaxPlusOne := 0x10000000000
  extcodecopy(someValidAddress, sizeTMaxPlusOne, 0, 32)
  return(0, 32)
}
```

as a write to memory address zero. The subsequent `return` then returns what was written by extcodecopy.

Reply:
Can you extract this into its own PR?

Reply:
Sure, found a mistake on this line in the process :-)

Created: https://github.com/ethereum/solidity/pull/12484